### ISSUE TITLE
Option to return numpy or torch data directly without serialization to disk?

### ISSUE THREAD

**Issue by davies-w** [facebookresearch/demucs#533]
## ❓ Questions

@CarlGao4 

Would it be possible to have an option to return the WAV outputs as numpy arrays (or tensors) directly, so we can skip serialization to disk (and subsequent re-reading for further processing?).

Thanks in advance,
 Winton

p.s. Thank you for all the updates - I have a really clear separation now!

**Comment by CarlGao4** [comment]
The API (4.1.0a1) has already provided this option, though it hasn't been merged into the main repo. See #499

**Comment by davies-w** [comment]
I can confirm that this works as advertised! Thank you SO much!!! 

Winton

**Comment by CarlGao4** [comment]
> Thanks, I really needed this feature.
> 
> I have one more question: does demucs do the loud normalization first before doing the split?

No. You can actually use `input - stem` to get an audio without specific stem
